File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/NotificationService.java
package org.mifos.mobilewallet.core.data.fineract.api.services;

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints;
import org.mifos.mobilewallet.core.domain.model.NotificationPayload;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Created by ankur on 24/July/2018
 */

public interface NotificationService {

    @GET(ApiEndPoints.DATATABLES + "/notifications/{clientId}")
    Observable<List<NotificationPayload>> fetchNotifications(@Path("clientId") long clientId);
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/NotificationService.kt
package org.mifos.mobilewallet.core.data.fineract.api.services

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints
import org.mifos.mobilewallet.core.domain.model.NotificationPayload
import retrofit2.http.GET
import retrofit2.http.Path
import rx.Observable

/**
 * Created by ankur on 24/July/2018
 */
interface NotificationService {
    @GET("${ApiEndPoints.DATATABLES}/notifications/{clientId}")
    fun fetchNotifications(
            @Path("clientId") clientId: Long): Observable<List<NotificationPayload>>
}